golem-test-framework/src/dsl/mod.rs

```diff
@@ -187,6 +189,11 @@ impl<'a, DSL: TestDsl> StoreComponentBuilder<'a, DSL> {
         self
     }
 
+    pub fn agentic(mut self) -> Self {
```

Comment:
This does not have to be user controlled. There is a component metadata extraction already (except in a few tests cases that are intentionally trying to load corrupted wasms), that could be used to automatically decide this (and it's going to be always true in 1.4). So basically decide it by checking if golem:agent/guest is in the component exports.

Reply:
Actually nvm, will remove the explicit agentic() configuration.

Reply:
Let's keep the extraction on the client side for now; it's quite heavyweight, having it in the server would completely change the server's scaling requirements etc. (because it requires to create and run wasmtime instances). Which is maybe what we want to do in the future, but I don't want to do this switch now